### PR TITLE
feat(done): changes needed for wasm-dpp integration

### PR DESCRIPTION
**Pull request by antouhou**
This pull request includes changes needed for the wasm-dpp integration

## Issue being fixed or feature implemented

Not all methods are in line with the old dpp interface

## What was done?

Changed signatures of several methods
Removed method chaining for setters on some data structures

## Breaking Changes

Some setters and getters now behave quite differently than before

## Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added or updated relevant unit/integration/functional/e2e tests
- [x] I have made corresponding changes to the documentation

**For repository code-owners and collaborators only**
- [ ] I have assigned this pull request to a milestone

### CODE REVIEW

**Review by qrayven — DISMISSED** [review]

**Review by qrayven — DISMISSED** [review]

File: dpp/src/identity/identity.rs [hunk 1]
```diff
@@ -254,15 +228,26 @@ impl Identity {
...
-        // // TODO identifier_default_deserializer: default deserializer should be changed to bytes
-        // // Identifiers fields should be replaced with the string format to deserialize Identity
         raw_object.replace_identifier_paths(IDENTIFIER_FIELDS_RAW_OBJECT, ReplaceWith::Base58)?;
 
         let identity: Identity = serde_json::from_value(raw_object)?;
 
         Ok(identity)
     }
 
+    pub fn from_raw_identity(raw_object: JsonValue) -> Result<Identity, ProtocolError> {
```
Comment (shumkov) [inline dpp/src/identity/identity.rs]: Why do we need both?

File: dpp/src/identity/identity_public_key.rs [hunk 2]
```diff
@@ -325,29 +320,13 @@ impl IdentityPublicKey {
...
-            }
-            _ => {
-                return Err(ProtocolError::ParsingError(format!(
-                    "the key length is invalid: {} Allowed sizes: 33 or 65 bytes",
-                    self.data.len()
-                )));
-            }
-        };
-        Ok(original_key.pubkey_hash().to_vec())
+        match self.data.len() {
```
Comment (shumkov) [inline dpp/src/identity/identity_public_key.rs]: Shouldn't it be based on type not on size?

**Review by shumkov — CHANGES_REQUESTED** [review]

File: dpp/src/identity/identity.rs (hunk 1 above)
Comment (antouhou) [reply dpp/src/identity/identity.rs]: That's an incorrect name - one for where ids and stuff are buffers, and the other one for when they are strings

File: dpp/src/identity/identity_public_key.rs (hunk 2 above)
Comment (antouhou) [reply dpp/src/identity/identity_public_key.rs]: Fixed

File: dpp/src/identity/identity.rs (hunk 1 above)
Comment (shumkov) [reply dpp/src/identity/identity.rs]: Then it should be from_json and from_raw

Comment (antouhou) [reply dpp/src/identity/identity.rs]: Fixed

**Review by shumkov — DISMISSED** [review]
👍

**Review by shumkov — DISMISSED** [review]
👍

**Review by shumkov — DISMISSED** [review]
👍

**Review by shumkov — APPROVED** [review]
Once again ))

**Pull request merged**